### PR TITLE
Update css variable tables

### PR DESCRIPTION
**Pull request by jschuler**
Closes https://github.com/patternfly/patternfly-react/issues/3521
Screen Shot 2020-04-20 at 12 01 23 PM

http://patternfly-org-v4.surge.sh/v4/

### CODE REVIEW

File: packages/gatsby-theme-patternfly-org/components/cssVariables/cssVariables.js [hunk 1]
```diff
@@ -1,105 +1,211 @@
...
+      applicableFiles,
+      rows: this.getFilteredRows(applicableFiles),
       sortBy: {
         index: 0,
-        direction: 'asc' // a-z
+        direction: "asc" // a-z
       }
     };
+
+    this.getFilteredRows = this.getFilteredRows.bind(this);
```
Comment (redallen) [inline packages/gatsby-theme-patternfly-org/components/cssVariables/cssVariables.js:95]: Why do we need to bind? Are class members not bound automagically? :(

File: packages/gatsby-theme-patternfly-org/components/cssVariables/cssVariables.js [hunk 2]
```diff
@@ -1,105 +1,211 @@
...
+          const passes =
+            searchRE === undefined ||
+            searchRE.test(selector) ||
+            searchRE.test(val.property) ||
+            searchRE.test(val.token) ||
+            searchRE.test(val.value) ||
+            (val.values && searchRE.test(JSON.stringify(val.values)));
+          if (passes) {
+            const rowKey = `${selector}_${val.property}`;
+            let cells = this.props.hideSelectorColumn ? [] : [selector];
```
Comment (redallen) [inline packages/gatsby-theme-patternfly-org/components/cssVariables/cssVariables.js:115]: Same thing with destructuring

File: packages/gatsby-theme-patternfly-org/components/cssVariables/cssVariables.js [hunk 3]
```diff
@@ -1,105 +1,211 @@
...
+    this.getFilteredRows = this.getFilteredRows.bind(this);
+    this.getDebouncedFiltedRows = this.getDebouncedFiltedRows.bind(this);
+    this.onCollapse = this.onCollapse.bind(this);
   }
 
-  onFilterChange = (_change, event) => {
+  getFilteredRows = (applicableFiles, searchRE) => {
+    let filteredRows = [];
+    let rowNumber = -1;
+    applicableFiles.forEach(file => {
```
Comment (redallen) [inline packages/gatsby-theme-patternfly-org/components/cssVariables/cssVariables.js:103]: There's a much prettier way to write this loop with `applicableFiles.flatMap(file => Object.entries(file).filter(searchRE).map(val => filteredRow)` if you're feeling up for it

File: packages/gatsby-theme-patternfly-org/components/cssVariables/cssVariables.js [hunk 4]
```diff
@@ -1,105 +1,211 @@
...
+  onCollapse(event, rowKey, isOpen) {
+    const { rows } = this.state;
+    rows[rowKey].isOpen = isOpen;
     this.setState({
-      filterValue: event.target.value
+      rows
     });
   }
 
+  getDebouncedFiltedRows = debounce(value => {
```
Comment (redallen) [inline packages/gatsby-theme-patternfly-org/components/cssVariables/cssVariables.js:176]: `getDebouncedFiltedRows` -> `getDebouncedFilteredRows`

**Comment by jschuler** [comment]
Thanks for the review @redallen , addressing your comments now

File: packages/gatsby-theme-patternfly-org/components/cssVariables/cssVariables.js (hunk 1 above)
Comment (jschuler) [reply packages/gatsby-theme-patternfly-org/components/cssVariables/cssVariables.js:115]: Not needed for the first 2, last one I changed the function to use arrow syntax so also not needed anymore, thanks

**Pull request merged**